doc/source/groupby.rst

```diff
@@ -1002,6 +1002,58 @@ See the :ref:`visualization documentation<visualization.box>` for more.
   to ``df.boxplot(by="g")``. See :ref:`here<visualization.box.return>` for
   an explanation.
 
+
+.. _groupby.pipe:
+
+Piping function calls
+~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 0.17.0
+
+Similar to the funcionality provided by ``DataFrames`` and ``Series``, functions that take ``GroupBy`` objects can be chained together using a ``pipe`` function to allow for a cleaner, more readable syntax.
+
+Imagine that one has a function f that both takes and returns a ``DataFrameGroupBy``, a function g that takes a ``DataFrameGroupBy`` and a ``DataFrame``, and a function h that takes a ``DataFrame`` and returns a number.  Imagine further that each of these takes a single argument.
+
+Instead of having to deeply compose these functions and their arguments like the following
+
+.. code-block:: python
+
+   >>> h(g(f(df.groupby('group'), arg1), arg2), arg4)
+
+one can write the following
+
+.. code-block:: python
+
+  >>> (df
+         .groupby('group')
+         .pipe(f, arg1)
+         .pipe(g, arg2)
+         .pipe(h, arg3))
+
+For example, imagine one had a DataFrame with columns 'A' and B'. Imagine a user wanted count the occurence of each value of column 'B' but in groups determined by columns 'A', and then the user wanted to apply further transformations on the result (say, by doubling each value).  One could write this as a pipeline of functions applied to the original dataframe:
+
+.. code-block:: python
+
+    def f(dfgb):
+        """
+        Take a DataFrameGroupBy and return a Series
+        of counts of the values of column B
+        """
+        return dfgb['B'].value_counts()
```

Comment:
I find this a bit a strange (not very useful) example, as this is executing an actual method of a DataFrameGroupBy method.